agent/consul/rate/handler.go

```diff
@@ -150,6 +155,7 @@ func NewHandler(cfg HandlerConfig, delegate HandlerDelegate) *Handler {
 		cfg:      new(atomic.Pointer[HandlerConfig]),
 		delegate: delegate,
 		limiter:  limiter,
+		logger:   logger.NamedIntercept("rate-limit"),
```

Comment:
I think sub-loggers will forward logs to the parent logger, not just their own registered sinks.

See here:
- https://github.com/hashicorp/go-hclog/blob/8b7499ad6ad46ce583e825d324d008032b644703/interceptlogger.go#L116
- https://github.com/hashicorp/go-hclog/blob/8b7499ad6ad46ce583e825d324d008032b644703/interceptlogger.go#L47

Which I think means we'd end up sending logs to STDOUT twice (and the dropper wouldn't be effective).

Maybe we need to create a totally new logger (perhaps with `Output: io.Discard`) and pass the parent logger as the sink argument in `NewLogDropSink`?

Reply:
Yes and that mean we need to change the drop log to accept a logger instead of a sink. I made the change, let me know if that what you had in mind.